### PR TITLE
Delete Slack Connector

### PR DESCRIPTION
**Pull request by lasryaric**
Implement the delete Slack connector logic.
Also Added the opportunity for the connector clean up function to use a database transaction.
Also added the transaction logic for Notion and a missing call to `NotionConnectorState.destroy()`

### CODE REVIEW

**Comment by lasryaric** [comment]
**Deployment instructions:**
Add these two vars env to the connectors service:
```
SLACK_CLIENT_ID=abc...
SLACK_CLIENT_SECRET=xyz...
```

File: connectors/src/api/delete_connector.ts [hunk 1]
```diff
@@ -73,21 +73,12 @@ const _deleteConnectorAPIHandler = async (
       });
     }
 
-    await connector.destroy();
+    await connector.destroy({ transaction: t });
 
     return res.json({
       success: true,
     });
-  } catch (e) {
```
Comment (fontanierh) [inline connectors/src/api/delete_connector.ts:81]: but then what happens if there is an error here ?

**Review by fontanierh — APPROVED** [review]
LGTM just one question, thanks for handling the notion part 🙏

File: connectors/src/api/delete_connector.ts (hunk 1 above)
Comment (fontanierh) [reply connectors/src/api/delete_connector.ts:81]: Didn't realize we had the top level handler ! SGTM !

**Pull request merged**